pkg/common/resources/resources_test.go

```diff
@@ -1701,3 +1701,29 @@ func TestIsEmpty(t *testing.T) {
 		})
 	}
 }
+
+func TestResource_DominantResource(t *testing.T) {
+	tests := []struct {
+		name     string
+		used     *Resource
+		capacity *Resource
+		wantName string
+	}{
+		{"nil receiver", nil, Zero, ""},
+		{"nil cap", Zero, nil, ""},
+		{"zero cap", NewResourceFromMap(map[string]Quantity{"A": 10}), Zero, ""},
+		{"over cap", NewResourceFromMap(map[string]Quantity{"A": 20}), NewResourceFromMap(map[string]Quantity{"A": 10}), "A"},
+		{"zero usage exist", NewResourceFromMap(map[string]Quantity{"A": 0}), NewResourceFromMap(map[string]Quantity{"A": 10}), "A"},
+		{"usage not in cap", NewResourceFromMap(map[string]Quantity{"B": 10}), NewResourceFromMap(map[string]Quantity{"A": 10}), ""},
+		{"multiple usages", NewResourceFromMap(map[string]Quantity{"B": 10, "A": 10}), NewResourceFromMap(map[string]Quantity{"A": 10, "B": 20}), "A"},
```

Comment:
nit:`{"multiple usages", NewResourceFromMap(map[string]Quantity{"A": 10, "B": 10}), NewResourceFromMap(map[string]Quantity{"A": 10, "B": 20}), "A"},`

Reply:
I did that on purpose to show that the map is random order